diff --git a/camera_pkg/src/camera_node.cpp b/camera_pkg/src/camera_node.cpp
index 3740686..be2aca0 100644
--- a/camera_pkg/src/camera_node.cpp
+++ b/camera_pkg/src/camera_node.cpp
@@ -17,7 +17,11 @@
 #include "rclcpp/rclcpp.hpp"
 #include "sensor_msgs/msg/image.hpp"
 #include "sensor_msgs/msg/compressed_image.hpp"
-#include "cv_bridge/cv_bridge.h"
+#if __has_include(<cv_bridge/cv_bridge.hpp>)
+#include <cv_bridge/cv_bridge.hpp>
+#elif __has_include(<cv_bridge/cv_bridge.h>)
+#include <cv_bridge/cv_bridge.h>
+#endif
 #include "deepracer_interfaces_pkg/srv/video_state_srv.hpp"
 #include "deepracer_interfaces_pkg/msg/camera_msg.hpp"
 #include "opencv2/opencv.hpp"
