diff --git a/servo_pkg/CMakeLists.txt b/servo_pkg/CMakeLists.txt
index 54488db..2e3d84d 100644
--- a/servo_pkg/CMakeLists.txt
+++ b/servo_pkg/CMakeLists.txt
@@ -13,6 +13,7 @@ endif()
 
 if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
   add_compile_options(-Wall -Wextra -Wpedantic)
+  add_compile_options(-Wno-deprecated-declarations)
 endif()
 
 # find dependencies
diff --git a/servo_pkg/src/servo_mgr.cpp b/servo_pkg/src/servo_mgr.cpp
index e50be4c..64de37a 100644
--- a/servo_pkg/src/servo_mgr.cpp
+++ b/servo_pkg/src/servo_mgr.cpp
@@ -68,7 +68,7 @@ namespace PWM {
     void ServoMgr::servoSubscriber(const deepracer_interfaces_pkg::msg::ServoCtrlMsg::SharedPtr servoMsg) {
         auto setPWM = [&](auto &servo, auto value, auto type) {
             if (value < -1.0 || value > 1.0) {
-                RCLCPP_ERROR(logger_, "Invalid servo request: %d", value);
+                RCLCPP_ERROR(logger_, "Invalid servo request: %f", value);
                 return;
             }
 
diff --git a/servo_pkg/src/servo_node.cpp b/servo_pkg/src/servo_node.cpp
index 631af08..439d645 100644
--- a/servo_pkg/src/servo_node.cpp
+++ b/servo_pkg/src/servo_node.cpp
@@ -45,6 +45,7 @@ int main(int argc, char **argv) {
     auto ledMgr = std::make_unique<PWM::LedMgr>(node->get_logger());
     auto qos = rclcpp::QoS(rclcpp::KeepLast(1));
     qos.best_effort();
+    auto qos_default = rclcpp::QoS(rclcpp::SystemDefaultsQoS()).get_rmw_qos_profile();
     auto servoMsgStrategy = std::make_shared<rclcpp::strategies::message_pool_memory_strategy::MessagePoolMemoryStrategy<deepracer_interfaces_pkg::msg::ServoCtrlMsg, 1>>();
     auto rawPWMMsgStrategy = std::make_shared<rclcpp::strategies::message_pool_memory_strategy::MessagePoolMemoryStrategy<deepracer_interfaces_pkg::msg::ServoCtrlMsg, 1>>();
 
@@ -62,7 +63,7 @@ int main(int argc, char **argv) {
                                                                                                  std::placeholders::_1),
                                                                                         rclcpp::SubscriptionOptions(),
                                                                                         rawPWMMsgStrategy);
-    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto servoCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>::SharedPtr servoCalService =
         node->create_service<deepracer_interfaces_pkg::srv::SetCalibrationSrv>(SET_CAL_SERVICE_NAME,
                                                                                std::bind(&PWM::ServoMgr::setCalHdl,
@@ -70,10 +71,10 @@ int main(int argc, char **argv) {
                                                                                std::placeholders::_1,
                                                                                std::placeholders::_2,
                                                                                std::placeholders::_3),
-                                                                               ::rmw_qos_profile_default,
+                                                                               qos_default,
                                                                                servoCalServiceCbGrp_);
 
-    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::MutuallyExclusive);                                                                                          
+    auto setGPIOServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::MutuallyExclusive);
     rclcpp::Service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>::SharedPtr setGPIOService =
         node->create_service<deepracer_interfaces_pkg::srv::ServoGPIOSrv>(SERVO_GPIO_SERVICE_NAME,
                                                                           std::bind(&PWM::ServoMgr::setGPIOHdl,
@@ -81,10 +82,10 @@ int main(int argc, char **argv) {
                                                                           std::placeholders::_1,
                                                                           std::placeholders::_2,
                                                                           std::placeholders::_3),
-                                                                          ::rmw_qos_profile_default,
+                                                                          qos_default,
                                                                           setGPIOServiceCbGrp_);
 
-    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getCalServiceCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>::SharedPtr getCalService =
         node->create_service<deepracer_interfaces_pkg::srv::GetCalibrationSrv>(GET_CAL_SERVICE_NAME,
                                                                        std::bind(&PWM::ServoMgr::getCalHdl,
@@ -92,10 +93,10 @@ int main(int argc, char **argv) {
                                                                        std::placeholders::_1,
                                                                        std::placeholders::_2,
                                                                        std::placeholders::_3),
-                                                                       ::rmw_qos_profile_default,
+                                                                       qos_default,
                                                                        getCalServiceCbGrp_);
 
-    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto setLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>::SharedPtr setLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::SetLedCtrlSrv>(SET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::setLedCtrlHdl,
@@ -103,10 +104,10 @@ int main(int argc, char **argv) {
                                                                            std::placeholders::_1,
                                                                            std::placeholders::_2,
                                                                            std::placeholders::_3),
-                                                                           ::rmw_qos_profile_default,
+                                                                           qos_default,
                                                                            setLedCtrlCbGrp_);
 
-    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::callback_group::CallbackGroupType::Reentrant);                                                                                          
+    auto getLedCtrlCbGrp_ = node->create_callback_group(rclcpp::CallbackGroupType::Reentrant);
     rclcpp::Service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>::SharedPtr getLedCtrlService =
         node->create_service<deepracer_interfaces_pkg::srv::GetLedCtrlSrv>(GET_LED_STATE_SERVICE_NAME,
                                                                            std::bind(&PWM::LedMgr::getLedCtrlHdl,
@@ -114,7 +115,7 @@ int main(int argc, char **argv) {
                                                                            std::placeholders::_1,
                                                                            std::placeholders::_2,
                                                                            std::placeholders::_3),
-                                                                           ::rmw_qos_profile_default,
+                                                                           qos_default,
                                                                            getLedCtrlCbGrp_);
 
     RCLCPP_INFO(node->get_logger(), "servo_node started");
