diff --git a/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_config.py b/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_config.py
index dd6670d..68f884a 100644
--- a/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_config.py
+++ b/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_config.py
@@ -25,7 +25,7 @@ GET_OTG_LINK_STATE_SERVICE_NAME = "get_otg_link_state"
 # OTG settings.
 
 ENABLE_OTG_PERIODIC_CHECK = True
-OTG_CHECK_PERIOD_IN_SECONDS = 2
+OTG_CHECK_PERIOD_IN_SECONDS = 5
 
-OTG_STATE_DIRECTORY = os.path.join(os.sep, "sys", "kernel", "debug", "dwc3.0.auto")
-OTG_LINK_STATE = "link_state"
+OTG_STATE_DIRECTORY = os.path.join(os.sep, "sys", "class", "net", "usb0")
+OTG_LINK_STATE = "operstate"
diff --git a/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_control_node.py b/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_control_node.py
index 481e4e5..444a4a4 100644
--- a/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_control_node.py
+++ b/src/aws-deepracer-systems-pkg/deepracer_systems_pkg/deepracer_systems_pkg/otg_module/otg_control_node.py
@@ -136,7 +136,7 @@ class OTGControlNode(Node):
            there is a chagne in the host connecton status.
         """
         with utility.AutoLock(self.otg_guard):
-            host_connected = "U0" in file_system_utils.read_line(
+            host_connected = "up" in file_system_utils.read_line(
                                                     os.path.join(otg_config.OTG_STATE_DIRECTORY,
                                                                  otg_config.OTG_LINK_STATE))
             if host_connected != self.otg_connected:
@@ -182,7 +182,6 @@ class OTGControlNode(Node):
         """
         # Setup connectivity from  Windows
         usb0 = False
-        usb1 = False
 
         # Setup connectivity from Windows
         if self.execute("ip link set usb0 up"):
@@ -194,16 +193,6 @@ class OTGControlNode(Node):
         if not usb0:
                 self.get_logger().error("Ethernet Over OTG enable failed for Windows.")
 
-        # Setup connectivity from Mac
-        if self.execute("ip link set usb1 up"):
-            if self.execute("ip addr add 10.0.1.1/30 dev usb1"):
-                if self.execute("systemctl restart dnsmasq") and \
-                   self.execute("systemctl restart isc-dhcp-server"):
-                    usb1 = True
-                    self.get_logger().info("Ethernet Over OTG enabled for MAC!")
-        if not usb1:
-                self.get_logger().error("Ethernet Over OTG enable failed for MAC.")
-
         return True
 
     def disable_otg(self):
@@ -211,8 +200,6 @@ class OTGControlNode(Node):
         """
         if not self.execute("ip link set usb0 down"):
             self.get_logger().error("Ethernet Over OTG disable failed for Windows!")
-        if not self.execute("ip link set usb1 down"):
-            self.get_logger().error("Ethernet Over OTG disable failed for MAC!")
         self.get_logger().info("Ethernet Over OTG disabled!!!")
 
     def execute(self, cmd):
